packages/hydrogen/src/analytics-manager/AnalyticsProvider.test.tsx

```diff
@@ -97,222 +97,267 @@ vi.mock('@remix-run/react', () => ({
   }),
 }));
 
+// Avoid downloading the PerfKit script in tests
+vi.mock('./PerfKit', () => ({
+  PerfKit: () => null,
+}));
+
 describe('<Analytics.Provider />', () => {
+  beforeAll(() => {
+    vi.stubGlobal(
+      'fetch',
+      function mockFetch(input: URL | RequestInfo): Promise<Response> {
+        const MONORAIL_ENDPOINT =
+          'https://monorail-edge.shopifysvc.com/unstable/produce_batch';
+        const CHECKOUT_ENDPOINT =
+          'https://checkout.hydrogen.shop/api/unstable/graphql.json';
+
+        if (input === MONORAIL_ENDPOINT || input === CHECKOUT_ENDPOINT) {
+          return Promise.resolve(
+            new Response('', {
+              status: 200,
+            }),
+          );
+        }
+
+        throw new Error('Analytics fetch mock - request not handled');
+      },
+    );
+  });
+
   it('renders its children', async () => {
     render(
       <Analytics.Provider cart={null} shop={SHOP_DATA} consent={CONSENT_DATA}>
         <div>child</div>;
       </Analytics.Provider>,
     );
+
+    // Wait until idle
+    await act(async () => {});
+
     expect(screen.getByText('child')).toBeInTheDocument();
   });
 
   describe('useAnalytics()', () => {
-    it('returns shop, cart, customData', () =>
-      new Promise((done) => {
-        renderAnalyticsProvider({
-          cart: CART_DATA,
-          customData: {test: 'test'},
-          callback: (analytics: AnalyticsContextValue | null) => {
-            expect(analytics?.canTrack()).toBe(true);
-            expect(analytics?.shop).toBe(SHOP_DATA);
-            expect(analytics?.cart).toBe(CART_DATA);
-            expect(analytics?.customData).toEqual({test: 'test'});
-            done(0);
-          },
```

Comment:
This had the assertions in a callback, and turns out the callback was running before all the rendering (and setTimeouts) were finished. I've changed this to be an awaitable Promise so that we can run assertions after it's done.

Reply:
Nice, much more readable as well.